### DISCUSSION TITLE
Manually detach/destroy

### DISCUSSION THREAD

**Discussion by eziP9AgWuixq** [weareseeed/react-square-web-payments-sdk#87] [Help]
is it possible to manually invoke the detach or destroy methods?

scenario:
the user is presented with a list of payment methods, once they select one, the rswp component renders that method along with a Back button. they click Back button to go back to the list of methods and choose the same one they selected the first time. this causes the rswp component to not render because it's already attached to the DOM.

when using the sdk straight from square, i would call a function to detach the element when the Back button was clicked. is there any way to handle this with the rswp package?

update: it seems this issue doesn't happen with Google Pay or Afterpay, but it does with Cashapp. with CreditCard, the flow mentioned above gives this error from Square: `Uncaught (in promise) PaymentMethodNotAttachedError: Card has not been attached to the page. It must be attached before you can call this method.`

**Comment by type-wolf** [comment]
You may be calling credit card methods without having completed initialisation
Not appropriate, but avoided by delaying the approximate time needed to initialise

```tsx
const PaymentForm: FC = () => {
  const [isMount, setIsMount] = useState<boolean>(false);

  useEffect(() => {
    /** Wait for 1 second for Square payment form initialisation */
    const timer = setTimeout(() => setIsMount(true), 1000);
    return () => {
      clearTimeout(timer);
      setIsMount(false);
    };
  }, []);

  return (
    <PaymentForm {...props}>
      {isMount ? ( <CreditCard />) : (<></>)}
    </PaymentForm >
  );
};

export default PaymentForm;
```
